### PR TITLE
FEAT: Add more POSTGIS operations

### PR DESCRIPTION
**Pull request by ian-r-rose**
Follow up to #1787 and #1925 . This adds a number of new operations for the PostGIS backend:

* `ST_GeometryType`
* `ST_GeometryN`
* `ST_IsValid`
* `ST_LineLocatePoint`
* `ST_LineMerge`
* `ST_LineSubstring`
* `ST_OrderingEquals`
* `ST_Union`

The last one requires special attention. That [`ST_Union`](http://postgis.net/docs/ST_Union.html) is available as both a value operation and a column aggregation operation. To avoid a name collision in the ibis APIs, I have named the aggregation variant `unary_union`. This is following the same choice made by [GeoPandas](http://geopandas.org/reference.html#geopandas.GeoSeries.unary_union).

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -1863,6 +1905,95 @@ def geo_intersects(left, right):
...
+
+    Returns
+    -------
+    valid : bool scalar
+    """
+    op = ops.GeoIsValid(arg)
+    return op.to_expr()
+
+
+def geo_line_locate_point(left, right):
```
Comment (xmnlab) [inline ibis/expr/api.py]:
```
ibis/expr/api.py:1925 in public function `geo_line_locate_point`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/api.py:1925 in public function `geo_line_locate_point`:
        D400: First line should end with a period (not 'e')
ibis/expr/api.py:1925 in public function `geo_line_locate_point`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
```

could you add a summary line here?

File: ibis/expr/api.py [hunk 2]
```diff
@@ -1863,6 +1905,95 @@ def geo_intersects(left, right):
...
+    Returns
+    -------
+    distance: float scalar
+    """
+    op = ops.GeoLineLocatePoint(left, right)
+    return op.to_expr()
+
+
+def geo_line_merge(arg):
+    """Returns a (set of) LineString(s) formed by sewing together the
```
Comment (xmnlab) [inline ibis/expr/api.py]:
```
ibis/expr/api.py:1943 in public function `geo_line_merge`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/api.py:1943 in public function `geo_line_merge`:
        D400: First line should end with a period (not 'e')
ibis/expr/api.py:1943 in public function `geo_line_merge`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
```

could you add a summary line here?

File: ibis/expr/api.py [hunk 3]
```diff
@@ -1863,6 +1905,95 @@ def geo_intersects(left, right):
...
+    Returns
+    -------
+    merged: geometry scalar
+    """
+    op = ops.GeoLineMerge(arg)
+    return op.to_expr()
+
+
+def geo_line_substring(arg, start, end):
+    """Return a linestring that is a substring of the input one, starting
```
Comment (xmnlab) [inline ibis/expr/api.py]:
```
ibis/expr/api.py:1961 in public function `geo_line_substring`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/api.py:1961 in public function `geo_line_substring`:
        D400: First line should end with a period (not 'g')
```

could you add a summary line here?

File: ibis/expr/api.py [hunk 4]
```diff
@@ -1863,6 +1905,95 @@ def geo_intersects(left, right):
...
+    Returns
+    -------
+    substring: linestring scalar
+    """
+    op = ops.GeoLineSubstring(arg, start, end)
+    return op.to_expr()
+
+
+def geo_ordering_equals(left, right):
+    """Returns true if the two geometries are equal and the coordinates
```
Comment (xmnlab) [inline ibis/expr/api.py]:
```
ibis/expr/api.py:1981 in public function `geo_ordering_equals`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/api.py:1981 in public function `geo_ordering_equals`:
        D400: First line should end with a period (not 's')
ibis/expr/api.py:1981 in public function `geo_ordering_equals`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
```

could you add a summary line here?

File: ibis/expr/api.py [hunk 5]
```diff
@@ -1967,6 +2098,42 @@ def geo_max_distance(left, right):
     return op.to_expr()
 
 
+def geo_unary_union(arg):
+    """Returns the pointwise union of the two geometries.
```
Comment (xmnlab) [inline ibis/expr/api.py]:
```
ibis/expr/api.py:2102 in public function `geo_unary_union`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/api.py:2102 in public function `geo_unary_union`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
```

could you add a summary line here?

File: ibis/expr/api.py [hunk 6]
```diff
@@ -1967,6 +2098,42 @@ def geo_max_distance(left, right):
...
+    -------
+    union : geometry scalar
+    """
+    expr = ops.GeoUnaryUnion(arg).to_expr()
+    expr = expr.name('union')
+    return expr
+
+
+def geo_union(left, right):
+    """Returns the pointwise union of the two geometries.
```
Comment (xmnlab) [inline ibis/expr/api.py]:
```
ibis/expr/api.py:2121 in public function `geo_union`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/api.py:2121 in public function `geo_union`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
```

could you add a summary line here?

File: ibis/expr/operations.py [hunk 7]
```diff
@@ -3074,6 +3088,55 @@ class GeoIntersects(GeoSpatialBinOp):
     output_type = rlz.shape_like('args', dt.boolean)
 
 
+class GeoIsValid(GeoSpatialUnOp):
+    """Returns true if the geometry is well-formed"""
```
Comment (xmnlab) [inline ibis/expr/operations.py]:
```
ibis/expr/operations.py:3092 in public class `GeoIsValid`:
        D400: First line should end with a period (not 'd')
```

could you add a period `.` at the end of the text?

File: ibis/expr/operations.py [hunk 8]
```diff
@@ -3074,6 +3088,55 @@ class GeoIntersects(GeoSpatialBinOp):
...
+
+    output_type = rlz.shape_like('args', dt.boolean)
+
+
+class GeoLineLocatePoint(GeoSpatialBinOp):
+    """Returns a float between zero and one representing the location of the
```
Comment (xmnlab) [inline ibis/expr/operations.py]:
```
ibis/expr/operations.py:3098 in public class `GeoLineLocatePoint`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/operations.py:3098 in public class `GeoLineLocatePoint`:
        D400: First line should end with a period (not 'e')
```

could you add a summary line here?

File: ibis/expr/operations.py [hunk 9]
```diff
@@ -3074,6 +3088,55 @@ class GeoIntersects(GeoSpatialBinOp):
...
+    """
+
+    left = Arg(rlz.linestring)
+    right = Arg(rlz.point)
+
+    output_type = rlz.shape_like('args', dt.halffloat)
+
+
+class GeoLineMerge(GeoSpatialUnOp):
+    """Returns a (set of) LineString(s) formed by sewing together the
```
Comment (xmnlab) [inline ibis/expr/operations.py]:
```
ibis/expr/operations.py:3110 in public class `GeoLineMerge`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/operations.py:3110 in public class `GeoLineMerge`:
        D209: Multi-line docstring closing quotes should be on a separate line
ibis/expr/operations.py:3110 in public class `GeoLineMerge`:
        D400: First line should end with a period (not 'e')
```

could you add a summary line here?

File: ibis/expr/operations.py [hunk 10]
```diff
@@ -3074,6 +3088,55 @@ class GeoIntersects(GeoSpatialBinOp):
...
+    constituent line work of a multilinestring. If a geometry other than
+    a linestring or multilinestring is given, this will return an empty
+    geometry collection."""
+
+    output_type = rlz.shape_like('args', dt.geometry)
+
+
+class GeoLineSubstring(GeoSpatialUnOp):
+    """Return a linestring that is a substring of the input one, starting
```
Comment (xmnlab) [inline ibis/expr/operations.py]:
```
ibis/expr/operations.py:3119 in public class `GeoLineSubstring`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/operations.py:3119 in public class `GeoLineSubstring`:
        D400: First line should end with a period (not 'g')
```

could you add a summary line here?

File: ibis/expr/operations.py [hunk 11]
```diff
@@ -3074,6 +3088,55 @@ class GeoIntersects(GeoSpatialBinOp):
...
+    arg = Arg(rlz.linestring)
+
+    start = Arg(rlz.floating)
+    end = Arg(rlz.floating)
+
+    output_type = rlz.shape_like('args', dt.linestring)
+
+
+class GeoOrderingEquals(GeoSpatialBinOp):
+    """Returns true if the two geometries are equal and the coordinates
```
Comment (xmnlab) [inline ibis/expr/operations.py]:
```
ibis/expr/operations.py:3134 in public class `GeoOrderingEquals`:
        D205: 1 blank line required between summary line and description (found 0)
ibis/expr/operations.py:3134 in public class `GeoOrderingEquals`:
        D209: Multi-line docstring closing quotes should be on a separate line
ibis/expr/operations.py:3134 in public class `GeoOrderingEquals`:
        D400: First line should end with a period (not 's')
```

could you add a summary line here?

**Comment by ian-r-rose** [comment]
Thanks for taking a look @xmnlab, I've made the requested changes to the docstrings.

**Pull request merged**